staging/src/k8s.io/apiextensions-apiserver/pkg/client/applyconfiguration/apiextensions/v1beta1/webhookclientconfig.go

```diff
@@ -51,9 +51,19 @@ func (b *WebhookClientConfigApplyConfiguration) WithService(value *ServiceRefere
 // WithCABundle adds the given value to the CABundle field in the declarative configuration
 // and returns the receiver, so that objects can be build by chaining "With" function invocations.
 // If called multiple times, values provided by each call will be appended to the CABundle field.
+// Deprecated: WithCABundle does not replace existing list for atomic list type. Use WithNewCABundle instead.
 func (b *WebhookClientConfigApplyConfiguration) WithCABundle(values ...byte) *WebhookClientConfigApplyConfiguration {
 	for i := range values {
 		b.CABundle = append(b.CABundle, values[i])
 	}
 	return b
 }
+
+// WithNewCABundle replaces the CABundle field in the declarative configuration with given values
+// and returns the receiver, so that objects can be build by chaining "With" function invocations.
+// If called multiple times, the CABundle field is set to the values of the last call.
+func (b *WebhookClientConfigApplyConfiguration) WithNewCABundle(values ...byte) *WebhookClientConfigApplyConfiguration {
+	b.CABundle = make([]byte, len(values))
+	copy(b.CABundle, values)
+	return b
```

Comment:
I actually - don't think we should do a deep copy here. 

Most controllers will create these configs in a reconcile loop and this is excessive

Reply:
Thanks for the feedback Dave, do you have concerns about performance specifically? what makes you think it's a problem? Do you think poeple will expect further changes to the values to match what they have in their config?

Reply:
> do you have concerns about performance specifically?  what makes you think it's a problem? 

Our webhook CAs rarely change - and performing this copy for every reconcile where we are generating the apply config seems unnecesary.

I'm assuming the outgoing request doesn't mutate the byte array. 

> Do you think poeple will expect further changes to the values to match what they have in their config?

Can you clarify your question?

Reply:
I'll agree that we probably don't need the copy:
1. caller can make the copy themselves if they want
2. the copy isn't deep anyway so the caller still needs to be aware and possibly properly copy.

Reply:
Hi @dprotaso I have some different opinion on this. This PR is meant to fix the With function for all atomic slice types. Even though it is true that byte slices rarely change, we cannot work under the same assumption for other types of slices. I think we should use `copy` to be safer for more generic slice types. Although it is probably excessive for bytes, I think the performance impact is probably tiny if the byte length is not excessive.